### PR TITLE
Cleanup doc repetitions in libcudf

### PR DESCRIPTION
**Pull request by karthikeyann**
## Description
Remove/update repeated documentation text
Remove declaration repetitions in tdigest.hpp

## Checklist
- [x] I am familiar with the [Contributing Guidelines](https://github.com/rapidsai/cudf/blob/HEAD/CONTRIBUTING.md).
- [x] New or existing tests cover these changes.
- [x] The documentation is up to date with these changes.

### CODE REVIEW

File: cpp/include/cudf/detail/utilities/vector_factories.hpp [hunk 1]
```diff
@@ -398,7 +398,7 @@ thrust::host_vector<typename Container::value_type> make_host_vector_async(
 }
 
 /**
- * @brief Synchronously construct a `std::vector` containing a copy of data from a
+ * @brief Synchronously construct a `thrust::vector` containing a copy of data from a
```
Comment (davidwendt) [inline cpp/include/cudf/detail/utilities/vector_factories.hpp:401]:
```suggestion
 * @brief Synchronously construct a `thrust::host_vector` containing a copy of data from a
```

File: cpp/include/cudf/detail/utilities/vector_factories.hpp [hunk 2]
```diff
@@ -417,7 +417,7 @@ thrust::host_vector<T> make_host_vector_sync(device_span<T const> v, rmm::cuda_s
 }
 
 /**
- * @brief Synchronously construct a `std::vector` containing a copy of data from a device
+ * @brief Synchronously construct a `thrust::vector` containing a copy of data from a device
```
Comment (karthikeyann) [inline cpp/include/cudf/detail/utilities/vector_factories.hpp:420]:
```suggestion
 * @brief Synchronously construct a `thrust::host_vector` containing a copy of data from a device
```

File: cpp/include/cudf/datetime.hpp [hunk 3]
```diff
@@ -78,7 +78,7 @@ std::unique_ptr<cudf::column> extract_day(
   rmm::mr::device_memory_resource* mr = rmm::mr::get_current_device_resource());
 
 /**
- * @brief  Extracts day from any datetime type and returns an int16_t
+ * @brief  Extracts week day from any datetime type and returns an int16_t
```
Comment (bdice) [inline cpp/include/cudf/datetime.hpp]: Weekday is one word.
```suggestion
 * @brief  Extracts weekday from any datetime type and returns an int16_t
```

File: cpp/include/cudf/detail/indexalator.cuh [hunk 4]
```diff
@@ -582,7 +582,7 @@ struct indexalator_factory {
   };
 
   /**
-   * @brief An index accessor that returns an index value if corresponding validity flag is true.
+   * @brief An index accessor that returns an index value if scalar's validity flag is true.
```
Comment (bdice) [inline cpp/include/cudf/detail/indexalator.cuh]:
```suggestion
   * @brief An index accessor that returns an index value if the scalar is valid.
```

File: cpp/include/cudf/io/orc_metadata.hpp [hunk 5]
```diff
@@ -355,13 +355,13 @@ class orc_metadata {
 };
 
 /**
- * @brief Reads file-level and stripe-level statistics of ORC dataset.
+ * @brief Reads meta data of ORC dataset.
```
Comment (bdice) [inline cpp/include/cudf/io/orc_metadata.hpp:358]:
```suggestion
 * @brief Reads metadata of ORC dataset.
```

File: cpp/include/cudf/datetime.hpp (hunk 3 above)
Comment (davidwendt) [inline cpp/include/cudf/datetime.hpp:81]:
```suggestion
 * @brief  Extracts a weekday from any datetime type and returns an int16_t
```

File: cpp/include/cudf/detail/indexalator.cuh (hunk 4 above)
Comment (davidwendt) [inline cpp/include/cudf/detail/indexalator.cuh:585]:
```suggestion
   * @brief An index accessor that returns an index value if the scalar's validity flag is true.
```

**Review by bdice — APPROVED** [review]
A couple minor suggestions. Otherwise LGTM.

**Review by davidwendt — APPROVED** [review]

**Comment by karthikeyann** [comment]
Note on other documentation string repetitions in libcudf.
There are many repetitions in files related to table, table_view, column, column_view, nvtx3.hpp, string_view.hpp/.cuh 
orc.hpp (writer/reader options, chunked options), parquet.hpp (writer/reader options, chunked options).
These repetitions are due to same functions repeated among the datatypes. Some could use inheritance to avoid repetition, and some are unavoidable. But they are not documentation changes. So, leaving them out of this PR.

**Pull request merged**